src/app/replayer/sql.ml

```diff
@@ -236,3 +236,35 @@ module Public_key = struct
   let run (module Conn : Caqti_async.CONNECTION) pk_id =
     Conn.find_opt query pk_id
 end
+
+module Epoch_ledger_hash = struct
+  (* given an epoch ledger hash, what are the state hashes in blocks
+     with a next epoch data id that refers to that epoch ledger hash
+
+     returned in descending order of the global slot in those blocks
+
+     the state hashes may or may not refer to blocks with a chain back
+     to the genesis block
+   *)
+
+  let query =
+    Caqti_request.collect Caqti_type.string Caqti_type.string
+      {|
+         SELECT state_hash FROM blocks
+
+         INNER JOIN epoch_data
+
+         ON blocks.next_epoch_data_id = epoch_data.id
+
+         INNER JOIN snarked_ledger_hashes
+
+         ON epoch_data.ledger_hash_id = snarked_ledger_hashes.id
+
+         WHERE snarked_ledger_hashes.value = ?
+
+         ORDER BY global_slot DESC
```

Comment:
Make this ascending order? The ledger that becomes the next epoch ledger is generated before the block that makes it the next epoch ledger. So the first block to have this as the next epoch ledger should be sufficient to look up the transactions

Reply:
If we use your suggested algorithm, we don't need to do the block lookup.